dist/js/multiselect.js

```diff
@@ -1,755 +1,1812 @@
+ * @property {jQuery} [$moveUp=$('#'+id+'_moveUp')] - the button to click to move the selected element(s) up one index
+ * @property {jQuery} [$moveDown=$('#'+id+'_moveDown')] - the button to click to move the selected element(s) down one index
+ */
+
+/**
+ * A wrapper for the part of the options where the user can provide
+ * jQuery selectors for the Multiselect elements
+ * @typedef {Object} ElementNames
+ * @property {jQuery} [right] - name of the element to become the right palette
+ * @property {jQuery} [leftAll] - name of the element to become the "Move all Left" button
+ * @property {jQuery} [rightAll] - name of the element to become the "Move all Right" button
+ * @property {jQuery} [leftSelected] - name of the element to become the "Move Selected Left" button
+ * @property {jQuery} [rightSelected] - name of the element to become the "Move Selected Right" button
+ * @property {jQuery} [undo] - name of the element to become the "Undo Last Move" button
+ * @property {jQuery} [redo] - name of the element to become the "Redo Last Undone Move" button
+ * @property {jQuery} [moveUp] - name of the element to become the "Move selected elements Up" button
+ * @property {jQuery} [moveDown] - name of the element to become the "Move selected elements Down" button
+ */
+
+/**
+ * Additional options wrapper for Multiselect that affects its behaviour
+ * @typedef {Object} MultiselectOptions
+ * @property {KeepInitialPositionFor} [keepRenderingFor=ALL] - what to keep the rendering order for
+ * @property {boolean} [submitAllLeft=true] - whether to submit all visible left options when a form is submitted
+ * @property {boolean} [submitAllRight=true] - whether to submit all visible right options when a form is submitted
+ * @property {SearchElements} [search] - the meta information used in the search elements
+ * @property {boolean} [ignoreDisabled=false] - whether to ignore disabled options when moving
+ * @property {string} [matchOptgroupBy='label'] - Which html attribute should be compared against when filtering
+ */
+
+/**
+ * Callback wrapper for the functions where Multiselect
+ * provides a way for the user to influence behaviour.
+ * @typedef {Object} CallbackFunctions
+ * @property {function} startUp - this function is called during initialisation, before the options are sorted
+ * @property {function} sort - used to define a different comparator from the default lexicographical comparison
+ * @property {function} beforeMoveToRight - called before the options are moved to the right side of the multiselect
+ * @property {function} afterMoveToRight - called after the options are moved to the right side of the multiselect
+ * @property {function} beforeMoveToLeft - called before the options are moved to the left side of the multiselect
+ * @property {function} afterMoveToLeft - called after the options are moved to the left side of the multiselect
+ * @property {function} move - called instead of the default move function
```

Comment:
The moveToLeft and moveToRight callbacks as well as the moveFromAToB callback was removed and the new move callback was provided. Later you can see that the things that must be done before and after move are still done. Before and after callbacks are still called, but now the user doesn't need to care about stacks or calling the after and before callbacks.